form.go

```diff
@@ -590,10 +602,19 @@ func (f *Form) Run() error {
 
 // run runs the form in normal mode.
 func (f *Form) run() error {
+	startTime := time.Now()
+	if f.timeout > 0 {
+		ctx, cancel := context.WithTimeout(context.Background(), f.timeout)
+		defer cancel()
+		f.teaOptions = append(f.teaOptions, tea.WithContext(ctx))
+	}
 	m, err := tea.NewProgram(f, f.teaOptions...).Run()
 	if m.(*Form).aborted {
 		err = ErrUserAborted
 	}
+	if err != nil && time.Since(startTime) >= f.timeout {
```

Comment:
Shouldn't this be `err == nil`? Also, every time I abort the form, I get a timeout error instead. This is because when `f.timeout` is zero, `time.Since(startTime)` will always be greater.

Reply:
It has to be `err != nil` because `tea.NewProgram` returns `ErrProgramKilled` . Though good catch!